server/src/main/java/org/apache/cassandra/sidecar/tasks/PeriodicTaskExecutor.java

```diff
@@ -130,22 +147,33 @@ private void schedule(PeriodicTaskKey key, long priorExecDurationMillis, long de
      */
     private void executeAndScheduleNext(PeriodicTaskKey key, long execCount)
     {
+        Promise<Void> runPromise = Promise.promise();
+        if (activeRuns.computeIfAbsent(key, k -> runPromise.future()) != runPromise)
```

Comment:
`activeRuns` no longer has any use for the _value_ of it's entries and can be converted to a `ConcurrentHashSet`, which simplifies this and `unschedule` quite a bit.

Reply:
Well, "quite a bit" is an overstatement but a bit.

Reply:
It still wants to grab the `future` of the active run, so that unschedule can chain "task.close()" after the run has finished, to avoid any interruption to the active run.

Reply:
But why do you need the `runPromise`? It's _never_ used for anything except to call `runPromise.complete()` on line 160 below - the future isn't ever actually used as far as I can tell. It _used_ to be used in `unschedule but now it's just ignored anyway - if you turn `activeRuns` into a `ConcurrentHashSet` and run all the tests, it just works (although that may be more of an indication that there are code paths that are uncovered in the current tests?)

Reply:
Added test to exercise that code path.